### ISSUE TITLE
intermittent auth hiesenbug: HTTPFailure: 401 - "The input authorization token can't serve the request"

### ISSUE THREAD

**Issue by RedCraig** [Azure/azure-cosmos-python#52]
I'm intermittently getting 401's when running tests:

``` bash
$ python test/crud_tests.py CRUDTests.test_attachment_crud_name_based
E
======================================================================
ERROR: test_attachment_crud_name_based (__main__.CRUDTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/crud_tests.py", line 1727, in test_attachment_crud_name_based
    self._test_attachment_crud(True);
  File "test/crud_tests.py", line 1843, in _test_attachment_crud
    media_response = client.ReadMedia(valid_attachment['media'])
  File "~/Projects/azure-documentdb-python/pydocumentdb/document_client.py", line 1566, in ReadMedia
    headers)
  File "~/Projects/azure-documentdb-python/pydocumentdb/document_client.py", line 2165, in __Get
    headers)
  File "~/Projects/azure-documentdb-python/pydocumentdb/synchronized_request.py", line 162, in SynchronizedRequest
    return _InternalRequest(connection_policy, request_options, request_body)
  File "~/Projects/azure-documentdb-python/pydocumentdb/synchronized_request.py", line 99, in _InternalRequest
    raise errors.HTTPFailure(response.status, data, headers)
HTTPFailure: Status code: 401
{"code":"Unauthorized","message":"The input authorization token can't serve the request. Please check
 that the expected payload is built as per the protocol, and check the key being used. Server used the 
following payload to sign: 'get\nmedia\nuo0hamswdgabaaaaaaaaaj3qfom=\nthu, 09 jun 2016 20:34:49 
gmt\n\n'\r\nActivityId: c0d2a470-5e3e-45f5-a4a1-0c2f05316540"}
```

Repeating the same test method a few times usually makes the above error occur.

Is there a way to get server logs for my docDB which will show the `ActivityId: c0d2a470-5e3e-45f5-a4a1-0c2f05316540"`?

I'm running latest master (this [commit](https://github.com/RedCraig/azure-documentdb-python/commit/fe95945cae62ee4a68ae53bc5519bec11f07522c)).

**Comment by RedCraig** [comment]
I feel like this is HMAC/timestamp related. How long is the HMAC timestamped request valid for before docDB server decides it's too old to honour?

**Comment by RedCraig** [comment]
Just realised this only occurs with the `test_attachment_crud_name_based` test case.

**Comment by southpolesteve** [comment]
Closing this issue due to age. Active development of our Python SDK has moved to https://github.com/Azure/azure-sdk-for-python. We have also released 4.0 which includes many updates and new features. If this issue still needs our attention, please reopen it in the new repository.
